superset/assets/src/visualizations/nvd3/PropTypes.js

```diff
@@ -0,0 +1,63 @@
+import PropTypes from 'prop-types';
+import { ANNOTATION_TYPES } from '../../modules/AnnotationTypes';
+
+export const numberOrAutoType = PropTypes.oneOfType([
+  PropTypes.number,
+  PropTypes.oneOf(['auto']),
+]);
+
+export const stringOrObjectWithLabelType = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.shape({
+    label: PropTypes.string,
+  }),
+]);
+
+export const colorObjectType = PropTypes.shape({
+  r: PropTypes.number.isRequired,
```

Comment:
good point. There were a few cases where it mysteriously breaks because some particular fields were omitted and I spent too much time figuring out. After found the cause, I was upset and add `isRequired` as warning, but this was not thoroughly verified for every field.